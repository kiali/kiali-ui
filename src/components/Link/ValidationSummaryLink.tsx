import * as React from 'react';
import IstioConfigListLink from './IstioConfigListLink';

interface Props {
  namespace: string;
  errors: number;
  warnings: number;
  objectCount?: number;
}

class ValidationSummaryLink extends React.PureComponent<Props> {
  hasIstioObjects = () => {
    return this.props.objectCount && this.props.objectCount > 0;
  };

  render() {
    let link: any = <div style={{ display: 'inline-block', marginLeft: '5px' }}>N/A</div>;

    if (this.hasIstioObjects()) {
      link = (
        <IstioConfigListLink
          namespaces={[this.props.namespace]}
          warnings={this.props.warnings > 0}
          errors={this.props.errors > 0}
        >
          {this.props.children}
        </IstioConfigListLink>
      );
    }

    return link;
  }
}

export default ValidationSummaryLink;
