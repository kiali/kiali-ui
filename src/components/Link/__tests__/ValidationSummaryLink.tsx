import * as React from 'react';
import { shallow } from 'enzyme';
import ValidationSummaryLink from '../ValidationSummaryLink';
import { shallowToJson } from 'enzyme-to-json';

const mockLink = (namespace: string, errors: number, warnings: number, objectCount?: number) => {
  return shallow(
    <ValidationSummaryLink namespace={namespace} errors={errors} warnings={warnings} objectCount={objectCount}>
      Summary Link
    </ValidationSummaryLink>
  );
};

describe('ValidationSummaryLink', () => {
  describe('when there are no objects', () => {
    it('is a N/A non-linked component', () => {
      const wrapper = mockLink('bookinfo', 0, 0, 0);

      expect(shallowToJson(wrapper)).toBeDefined();
      expect(shallowToJson(wrapper)).toMatchSnapshot();

      expect(wrapper.html()).toContain('N/A');
    });
  });

  describe('when there are objects', () => {
    describe('and only has errors', () => {
      it('points to IstioConfig list with errors filter', () => {
        const wrapper = mockLink('bookinfo', 2, 0, 2);

        expect(shallowToJson(wrapper)).toBeDefined();
        expect(shallowToJson(wrapper)).toMatchSnapshot();

        const linkComponent = wrapper.find('IstioConfigListLink');
        expect(linkComponent).toBeDefined();

        const linkProps = linkComponent.props();
        expect(linkProps.children).toContain('Summary Link');
        expect(linkProps['errors']).toBeTruthy();
        expect(linkProps['warnings']).toBeFalsy();
        expect(linkProps['namespaces']).toEqual(['bookinfo']);
      });
    });

    describe('and only has warnings', () => {
      it('points to IstioConfig list with warnings filter', () => {
        const wrapper = mockLink('bookinfo', 0, 3, 2);

        expect(shallowToJson(wrapper)).toBeDefined();
        expect(shallowToJson(wrapper)).toMatchSnapshot();

        const linkComponent = wrapper.find('IstioConfigListLink');
        expect(linkComponent).toBeDefined();

        const linkProps = linkComponent.props();
        expect(linkProps.children).toContain('Summary Link');
        expect(linkProps['errors']).toBeFalsy();
        expect(linkProps['warnings']).toBeTruthy();
        expect(linkProps['namespaces']).toEqual(['bookinfo']);
      });
    });

    describe('and has both errors and warnings', () => {
      it('points to IstioConfig list with both warnings and errors filter', () => {
        const wrapper = mockLink('bookinfo', 2, 3, 2);

        expect(shallowToJson(wrapper)).toBeDefined();
        expect(shallowToJson(wrapper)).toMatchSnapshot();

        const linkComponent = wrapper.find('IstioConfigListLink');
        expect(linkComponent).toBeDefined();

        const linkProps = linkComponent.props();
        expect(linkProps.children).toContain('Summary Link');
        expect(linkProps['errors']).toBeTruthy();
        expect(linkProps['warnings']).toBeTruthy();
        expect(linkProps['namespaces']).toEqual(['bookinfo']);
      });
    });
  });
});
