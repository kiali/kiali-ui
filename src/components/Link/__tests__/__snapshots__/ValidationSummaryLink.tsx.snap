// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationSummaryLink when there are no objects is a N/A non-linked component 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "marginLeft": "5px",
    }
  }
>
  N/A
</div>
`;

exports[`ValidationSummaryLink when there are objects and has both errors and warnings points to IstioConfig list with both warnings and errors filter 1`] = `
<IstioConfigListLink
  errors={true}
  namespaces={
    Array [
      "bookinfo",
    ]
  }
  warnings={true}
>
  Summary Link
</IstioConfigListLink>
`;

exports[`ValidationSummaryLink when there are objects and only has errors points to IstioConfig list with errors filter 1`] = `
<IstioConfigListLink
  errors={true}
  namespaces={
    Array [
      "bookinfo",
    ]
  }
  warnings={false}
>
  Summary Link
</IstioConfigListLink>
`;

exports[`ValidationSummaryLink when there are objects and only has warnings points to IstioConfig list with warnings filter 1`] = `
<IstioConfigListLink
  errors={false}
  namespaces={
    Array [
      "bookinfo",
    ]
  }
  warnings={true}
>
  Summary Link
</IstioConfigListLink>
`;
